File: src/test/java/controllers/APITests.java
package controllers;

import com.intuit.karate.junit5.Karate;
import in.specmatic.stub.ContractStub;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.springframework.boot.SpringApplication;
import org.springframework.context.ConfigurableApplicationContext;

import java.io.IOException;

import static in.specmatic.stub.API.createStub;

public class APITests {

    public static final String SPECMATIC_STUB_HOST = "localhost";
    public static final int SPECMATIC_STUB_PORT = 9000;
    private static ConfigurableApplicationContext context;
    private static ContractStub stub;

    @Karate.Test
    public Karate runTests() {
        return new Karate().relativeTo(APITests.class).path("apiTests.feature");
    }

    @BeforeAll
    public static void setUp() {
        context = SpringApplication.run(Application.class);
        stub = createStub(SPECMATIC_STUB_HOST, SPECMATIC_STUB_PORT);
    }

    @AfterAll
    public static void tearDown() throws IOException {
        if (context != null) {
            context.close();
        }
        if (stub != null) {
            stub.close();
        }
    }
}

File: src/test/kotlin/controllers/APITests.kt
package controllers

import `in`.specmatic.stub.ContractStub
import `in`.specmatic.stub.createStub
import com.intuit.karate.junit5.Karate
import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.BeforeAll
import org.springframework.boot.SpringApplication
import org.springframework.context.ConfigurableApplicationContext

class APITests {
    @Karate.Test
    fun apiTests(): Karate {
        return Karate().path("apiTests.feature").relativeTo(this::class.java)
    }

    companion object {
        private var service: ConfigurableApplicationContext? = null

        private lateinit var stub: ContractStub

        @BeforeAll
        @JvmStatic
        fun setUp() {
            stub = createStub()
            service = SpringApplication.run(Application::class.java)
        }

        @AfterAll
        @JvmStatic
        fun tearDown() {
            service?.stop()
            stub.close()
        }
    }
}